WooCommerce/Classes/ViewRelated/Orders/OrderListViewModel.swift

```diff
@@ -297,10 +305,13 @@ final class OrderListViewModel {
             let IPPresultsCount = IPPTransactionsFound.count
 
             if !hasResults {
+                inPersonPaymentsFeedbackCampaignTracked(campaign: .inPersonPaymentsCashOnDelivery)
                 return .IPP_COD
             } else if IPPresultsCount < Constants.numberOfTransactions {
+                inPersonPaymentsFeedbackCampaignTracked(campaign: .inPersonPaymentsFirstTransaction)
                 return .IPP_firstTransaction
             } else if IPPresultsCount >= Constants.numberOfTransactions {
+                inPersonPaymentsFeedbackCampaignTracked(campaign: .inPersonPaymentsPowerUsers)
```

Comment:
If I understand this correctly, the `shown` event will be tracked every time we call `feedbackBannerSurveySource` to read the source value. Right now, this is called only from `viewDidLoad`. But in future, if we call `feedbackBannerSurveySource()` from other places to read the source for a different purpose, it will register unnecessary tracks events.

What do you think about introducing a separate method just for tracking the `shown` event and calling it from the view controller? (Maybe from [this line](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L799) where we actually show the alert)

Reply:
At the moment I'd rather prefer to leave it as it is for the following reasons ( unless you have any concern, of course!)

- While we inform we'll be loading the survey banner in `viewDidLoad`, we need to wait for different results controllers in the model to return which content and survey link we'll be using. There's some async work as well tied to `syncIPPBannerVisibility` that we need to perform before fetching the feedback status. Right now we're assuring that we the correct campaign is passed to analytics as soon as we calculate which one we must use.
- The idea is to remove this survey in the following months and use something more robust, like JITM

In any case, please let me know if I miss-understood. Happy to discuss further!

Reply:
Thanks for the details @iamgabrielma! 🙇  Look like there is definitely more context to this that I am missing. 😅 I think someone from Peacock will be able to review this part better. 

> Right now we're assuring that we the correct campaign is passed to analytics as soon as we calculate which one we must use.

If we must do the tracking right after we calculate `SurveyViewController.Source`, instead of calling tracking method from `feedbackBannerSurveySource()` can we call the tracking method from the viewController right after [this line](https://github.com/woocommerce/woocommerce-ios/blob/cfe315f6c3124a99cfe7fd05bed167c7e38fb77c/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L165)?

This way we can make sure that the event is tracked only upon `viewDidLoad` and not if we accidentally call `feedbackBannerSurveySource()` again from a different place. 

WDYT?

Reply:
Oh nice, that alternative seems much better indeed! Changed on 18990a6aed8746be251bee3d24d5afad2cb43d29